code/go/0chain.net/smartcontract/dbs/event/event.go

```diff
@@ -61,16 +62,16 @@ func (edb *EventDb) GetEvents(ctx context.Context, block int64) ([]Event, error)
 }
 
 func (edb *EventDb) exists(ctx context.Context, event Event) (bool, error) {
-	var count int64
-	result := edb.Store.Get().WithContext(ctx).
-		Model(&Event{}).
-		Where("tx_hash = ? AND index = ?", event.TxHash, event.Index).
-		Count(&count)
+	var ev Event
+	result := edb.Store.Get().WithContext(ctx).Model(&Event{}).Where(&Event{TxHash: event.TxHash, Index: event.Index}).Take(&ev)
```

Comment:
shouldn't we be checking type and tag here as well @Sriep

Reply:
done